File: android-core/src/test/java/com/mparticle/BaseEventTest.java
package com.mparticle;

import org.junit.Test;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

public class BaseEventTest {

    @Test
    public void testEventType() {
        BaseEvent baseEvent = new BaseEvent(BaseEvent.Type.COMMERCE_EVENT);
        assertEquals(BaseEvent.Type.COMMERCE_EVENT, baseEvent.getType());
    }

    @Test
    public void testCustomFlags() {
        BaseEvent baseEvent = new BaseEvent(BaseEvent.Type.BREADCRUMB);
        assertNull(baseEvent.getCustomFlags());

        List<String> values1 = new ArrayList<String>();
        values1.add("val1");
        values1.add("val2");
        values1.add("val3");

        List<String> values2 = new ArrayList<String>();
        values2.add("val2");

        Map<String, List<String>> customFlags = new HashMap<String, List<String>>();
        customFlags.put("key1", values1);
        customFlags.put("key2", values2);
        customFlags.put("key3", new ArrayList<String>());
        baseEvent.setCustomFlags(customFlags);

        //should not be able to add null key
        customFlags.put(null, new ArrayList<String>());
        baseEvent.setCustomFlags(customFlags);
        assertEquals(3, baseEvent.getCustomFlags().size());

        baseEvent.setCustomFlags(null);
        assertNull(baseEvent.getCustomFlags());
    }

}

File: android-core/src/test/kotlin/com/mparticle/BaseEventTest.kt
package com.mparticle

import org.junit.Assert
import org.junit.Test

class BaseEventTest {
    @Test
    fun testEventType() {
        val baseEvent = BaseEvent(BaseEvent.Type.COMMERCE_EVENT)
        Assert.assertEquals(BaseEvent.Type.COMMERCE_EVENT, baseEvent.type)
    }

    @Test
    fun testCustomFlags() {
        val baseEvent = BaseEvent(BaseEvent.Type.BREADCRUMB)
        Assert.assertNull(baseEvent.customFlags)
        val values1: MutableList<String> = ArrayList()
        values1.add("val1")
        values1.add("val2")
        values1.add("val3")
        val values2: MutableList<String> = ArrayList()
        values2.add("val2")
        val customFlags = HashMap<String?, List<String>>()
        customFlags["key1"] = values1
        customFlags["key2"] = values2
        customFlags["key3"] = ArrayList()
        baseEvent.customFlags = customFlags

        // should not be able to add null key
        customFlags[null] = ArrayList()
        baseEvent.customFlags = customFlags
        Assert.assertEquals(3, baseEvent.customFlags?.size)
        baseEvent.customFlags = null
        Assert.assertNull(baseEvent.customFlags)
    }
}
